File: 1936.新增的最少台阶数.py
#
# @lc app=leetcode.cn id=1936 lang=python3
#
# [1936] 新增的最少台阶数
#

# @lc code=start
class Solution:
    def addRungs(self, rungs: List[int], dist: int) -> int:
        res = 0
        s = 0
        for i in range(len(rungs)):
            if rungs[i] - s > dist:
                res += (rungs[i] - s - 1) // dist
            s = rungs[i]
        return res
# @lc code=end

File: 1936.新增的最少台阶数.go
/*
 * @lc app=leetcode.cn id=1936 lang=golang
 *
 * [1936] 新增的最少台阶数
 */

// @lc code=start
func addRungs(rungs []int, dist int) int {
	res, s := 0, 0
	for _, r := range rungs {
		if r-s > dist {
			res += (r - s - 1) / dist
		}
		s = r
	}
	return res

}

// @lc code=end

